runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,403 @@
+
+/**
+ * A client to interface with log replication utilizing the logical groups replication model.
+ */
+@Slf4j
+public class LogReplicationLogicalGroupClient {
+
+    /**
+     * Key: ClientRegistrationId
+     * Value: ClientRegistrationInfo
+     */
+    public static final String LR_REGISTRATION_TABLE_NAME = "LogReplicationRegistrationTable";
+
+    /**
+     * Key: ClientDestinationInfoKey
+     * Value: DestinationInfoVal
+     */
+    public static final String LR_MODEL_METADATA_TABLE_NAME = "LogReplicationModelMetadataTable";
+
+    private static final ReplicationModel model = ReplicationModel.LOGICAL_GROUPS;
+
+    private final CorfuStore corfuStore;
+    private final Table<ClientRegistrationId, ClientRegistrationInfo, Message> replicationRegistrationTable;
+    private final Table<ClientDestinationInfoKey, DestinationInfoVal, Message> sourceMetadataTable;
+
+    private final ClientRegistrationId clientKey;
+    private final ClientRegistrationInfo clientInfo;
+
+    private final String clientName;
+
+    /**
+     * Constructor for LogReplicationLogicalGroupClient
+     *
+     * @param runtime Corfu Runtime
+     * @param clientName String representation of the client name
+     * @throws InvocationTargetException InvocationTargetException
+     * @throws NoSuchMethodException NoSuchMethodException
+     * @throws IllegalAccessException IllegalAccessException
+     */
+    public LogReplicationLogicalGroupClient(CorfuRuntime runtime, String clientName) throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+        Preconditions.checkArgument(isValid(clientName), String.format("%s", "clientName is null or empty."));
+        Table<ClientDestinationInfoKey, DestinationInfoVal, Message> tempSourceMetadataTable;
+        Table<ClientRegistrationId, ClientRegistrationInfo, Message> tempReplicationRegistrationTable;
+        this.corfuStore = new CorfuStore(runtime);
+        this.clientName = clientName;
+
+        this.clientKey = ClientRegistrationId.newBuilder()
+                .setClientName(clientName)
+                .build();
+        Instant time = Instant.now();
+        this.clientInfo = ClientRegistrationInfo.newBuilder()
+                .setClientName(clientName)
+                .setModel(model)
+                .setRegistrationTime(Timestamp.newBuilder().setSeconds(time.getEpochSecond()).setNanos(time.getNano()).build())
+                .build();
+
+        try {
+            tempReplicationRegistrationTable = corfuStore.getTable(CORFU_SYSTEM_NAMESPACE, LR_REGISTRATION_TABLE_NAME);
+        } catch (NoSuchElementException nse) {
+            tempReplicationRegistrationTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LR_REGISTRATION_TABLE_NAME, ClientRegistrationId.class,
+                    ClientRegistrationInfo.class, null, TableOptions.fromProtoSchema(ClientRegistrationInfo.class));
+        }
+        this.replicationRegistrationTable = tempReplicationRegistrationTable;
```

Comment:
I saw this change was from the comment https://github.com/CorfuDB/CorfuDB/pull/3463#discussion_r1067352764
but I have a question here about how could it benefit us: the `getTable()` could help avoid recreation of `Table` object when the client is using the same CorfuRuntime to register, otherwise we still have to do the `openTable`. 
but would multiple clients using the same runtime to register? I was supposing they would even be in different JVMs right?
@pankti-m please correct me if I'm wrong...